### PR TITLE
Use updated ECR containerd resolver

### PR DESCRIPTION
**Pull request by jahkeup**
**Issue number:**

Supporting change for #468 

**Description of changes:**

This uses a ~~unmerged~~ newly landed feature in [`amazon-ecr-containerd-resolver` that handles multi-arch manifest lists](https://github.com/awslabs/amazon-ecr-containerd-resolver/pull/13). It also allows for the removal of some duplicated URI logic that's been eliminated here.

Some at-hand lints were addressed during removal of unused code.

**Testing done:**

Unit tests continue to pass.

Hand testing builds with the multi-arch support at runtime - will post results.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by etungsten — APPROVED** [review]
:tea:

File: sources/host-ctr/cmd/host-ctr/main.go [hunk 1]
```diff
@@ -221,63 +219,82 @@ func _main() int {
...
+		cleanup, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+		_, err := task.Delete(cleanup)
+		if err != nil {
+			log.G(cleanup).WithError(err).Error("Failed to delete container task")
+		}
+	}()
 
-	// Wait before calling start in case the container task finishes too quickly
+	// Wait to call start in case the container task finishes too quickly.
```
Comment (samuelkarp) [inline sources/host-ctr/cmd/host-ctr/main.go]: The previous comment is actually more correct.  `task.Wait` is roughly patterned after `wait(2)` in that you'll receive notification when the task exits.  If the task exits before you call `Wait`, you don't get to read its status.  By calling `Wait` before `Start`, you guarantee that you receive the notification.

We're not delaying the start, but monitoring.

Comment (jahkeup) [reply sources/host-ctr/cmd/host-ctr/main.go]: Ah, yep I read this, as you pointed out, saying it is a *delayed* call - I'll fix the comment.

File: sources/host-ctr/cmd/host-ctr/main.go [hunk 2]
```diff
@@ -107,44 +132,60 @@ func _main() int {
 		return 1
 	}
 
-	// If the image is from ECR, the image reference will be converted into the form of
-	// `"ecr.aws/" + the ARN of the image repository + label/digest`.
-	// We tag the image with its original image name so other services can discover this image by its original image reference.
-	// After the tag operation, this image should be addressable by both its original image reference and its ECR resolver resolvable reference.
-	if match {
-		// Include original tag on ECR image for other consumers.
+	// When the image is from ECR, the image reference will be converted its ref
```
Comment (webern) [inline sources/host-ctr/cmd/host-ctr/main.go]: Possible typo at `will be converted its ref`.

**Review by webern — APPROVED** [review]
👍

**Review by samuelkarp — APPROVED** [review]

**Pull request merged**